src/HomeChainCoordinator.sol

```diff
@@ -1,229 +1,311 @@
 import {BitcoinUtils} from "./libraries/BitcoinUtils.sol";
-import {PSBTMetadata, IHomeChainCoordinator} from "./interfaces/IHomeChainCoordinator.sol";
 import {BitcoinLightClient} from "./BitcoinLightClient.sol";
+import {PSBTData, IHomeChainCoordinator} from "./interfaces/IHomeChainCoordinator.sol";
+import {Origin, IBaseChainCoordinator} from "./interfaces/IBaseChainCoordinator.sol";
 
 /**
  * @title HomeChainCoordinator
- * @dev Contract for coordinating cross-chain messages on the home chain
+ * @notice Contract for coordinating cross-chain messages on the home chain
+ * @dev Handles verification and cross-chain messaging for Bitcoin transactions
  */
 contract HomeChainCoordinator is OApp, ReentrancyGuard, Pausable, IHomeChainCoordinator {
     // Errors
     error TxnAlreadyProcessed(bytes32 btcTxnHash);
     error InvalidPSBTData();
     error InvalidAmount(uint256 amount);
-    error InvalidDestination();
+    error InvalidSourceOrDestination();
     error InvalidReceiver(address receiver);
     error BitcoinTxnNotFound();
     error BitcoinTxnAndPSBTMismatch();
     error InvalidPeer();
     error WithdrawalFailed();
+    error InvalidStatusUpdate();
+    error InvalidRequest();
+
+    struct StoreMessageParams {
+        bool txnType; // Whether the transaction is a mint or burn
+        bytes32 blockHash; // The block hash for the transaction
+        bytes32 btcTxnHash; // The BTC transaction hash
+        bytes32[] proof; // The proof for the transaction
+        uint256 index; // The index of the transaction in the block
+        bytes rawTxn; // Raw hex PSBT data for the mint or burn transaction
+        bytes32 taprootAddress; // The taproot address for the transaction
+        bytes32 networkKey; // The network key for the AVS
+        address[] operators; // The operators for the AVS
+    }
 
     // State variables
-    BitcoinLightClient private lightClient;
+    BitcoinLightClient internal immutable _lightClient;
+    uint32 internal immutable _chainEid;
 
-    address private taskManager;
-    mapping(bytes32 => PSBTMetadata) private btcTxnHash_psbtMetadata;
+    mapping(bytes32 => PSBTData) internal _btcTxnHash_psbtData;
 
     uint256 public maxGasTokenAmount = 1 ether; // Max amount that can be put as the native token amount
     uint256 public minBTCAmount = 1000; // Min BTC amount / satoshis that needs to be locked
 
+    bytes internal constant OPTIONS = hex"0003010011010000000000000000000000000000c350";
+
     // Events
-    event MessageSent(uint32 indexed dstEid, bytes32 indexed psbtHash, address indexed operator, uint256 timestamp);
-    event OperatorStatusChanged(address indexed operator, bool status);
+    event MessageCreated(bool indexed txnType, bytes32 indexed blockHash, bytes32 indexed btcTxnHash);
+    event MessageSent(uint32 indexed dstEid, bytes32 indexed btcTxnHash, address indexed operator, uint256 timestamp);
+    event MessageReceived(
+        bytes32 indexed guid,
+        uint32 srcEid,
```

Comment:
always try to put all indexed parameters first in the list, It always helps the off-chain code to filter based on indexed parameters in a generic fashion otherwise, Rahul has to write a filter for every event specifically.